### PR TITLE
Removing Flink adapter in favor of the one included with Flink

### PR DESCRIPTION
**Pull request by dlg99**
### Motivation

Removing Flink adapter in favor of https://github.com/apache/flink/tree/master/flink-connectors/flink-connector-pulsar
This one is based on an old version of Flink which brings in dependencies with various CVEs, since that version Flink added pulsar connector in their project. 

### Modifications

Removed Flink adapter, tests, examples, and dependencies.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): *YES*, removed
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? removes the feature
  - If yes, how is the feature documented? not applicable but should be mentioned in the release notes
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: examples/kafka-streams/src/main/java/org/apache/kafka/kafkastreams/pulsar/example/PageViewTypedDemo.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+ * To observe the results, read the output topic (e.g., via bin/kafka-console-consumer)
+ * - Topic: streams-pageviewstats-typed-output
+ *   Key Format: (JSON) {"_t": "wpvbr", "windowStart": (long ms WINDOW_TIMESTAMP), "region": (String REGION)}
+ *   Value Format: (JSON) {"_t": "rc", "count": (long REGION_COUNT), "region": (String REGION)}
+ *
+ * Note, the "_t" field is necessary to help Jackson identify the correct class for deserialization in the
+ * generic {@link JSONSerde}. If you instead specify a specific serde per class, you won't need the extra "_t" field.
+ */
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class PageViewTypedDemo {
```
Comment (michaeljmarshall) [inline examples/kafka-streams/src/main/java/org/apache/kafka/kafkastreams/pulsar/example/PageViewTypedDemo.java:85]: @dlg99 - is this kafka work related to removing flink support? I am not finding a connection, but I could be missing something.

**Review by eolivelli — APPROVED** [review]
+1

@merlimat

**Comment by merlimat** [comment]
We could also update the README to mention the adapters that are now developed somewhere else

File: examples/kafka-streams/src/main/java/org/apache/kafka/kafkastreams/pulsar/example/PageViewTypedDemo.java (hunk 1 above)
Comment (dlg99) [reply examples/kafka-streams/src/main/java/org/apache/kafka/kafkastreams/pulsar/example/PageViewTypedDemo.java:85]: oh no, I'll fix that

Comment (michaeljmarshall) [reply examples/kafka-streams/src/main/java/org/apache/kafka/kafkastreams/pulsar/example/PageViewTypedDemo.java:85]: Great, thanks. Other than this point, this LGTM.

Comment (dlg99) [reply examples/kafka-streams/src/main/java/org/apache/kafka/kafkastreams/pulsar/example/PageViewTypedDemo.java]: Fixed

**Comment by dlg99** [comment]
@merlimat I updated readme

**Review by michaeljmarshall — APPROVED** [review]

**Review by merlimat — APPROVED** [review]

**Review by aahmed-se — APPROVED** [review]
